test/tools/base64_test.cpp

```diff
@@ -18,69 +18,245 @@
+
+    std::string decodedStr = uprotocol::tools::Base64::base64decode(encodeStr);
+    ASSERT_EQ(decodedStr, origStr);
+
+    std::string decodedStrNoPadding = uprotocol::tools::Base64::base64decode(encodeStrNoPadding);
+    ASSERT_EQ(decodedStrNoPadding, origStr);
+};
+
+/**
+* Test Case: Tests single character input data
+*/
+TEST_F(Base64EncodeDecodeTests, input_SingleChar){
+    std::string origStr = "a";
+    std::string encodedStr = uprotocol::tools::Base64::base64encode(origStr);
+    std::string decodedStr = uprotocol::tools::Base64::base64decode(encodedStr);
+    ASSERT_EQ(decodedStr, origStr);
+};
+
+/**
+* Test Case: Tests empty input data
+*/
+TEST_F(Base64EncodeDecodeTests, emptyInputData){
+    std::string origStr = "";
+    std::string encodedStr = uprotocol::tools::Base64::base64encode(origStr);
+    std::string decodedStr = uprotocol::tools::Base64::base64decode(encodedStr);
+    ASSERT_EQ(decodedStr, origStr);
+};
+
+/**
+* Test Case: Tests Base64 encode and decode when passing invalid data length
+*/
+TEST_F(Base64EncodeDecodeTests, invalid_encoded_decoded_datalength){
+   
+    static const char origStr[] = "abcdefg";
+    size_t origStrLen = sizeof(origStr);
+    size_t origStrInvalidLen = 5;
+    ASSERT_NE(origStrLen, origStrInvalidLen);
+
+    std::string encodedStrVal = uprotocol::tools::Base64::base64encode((uint8_t *)origStr,
+                                                                origStrLen);
+    std::string encodedStrInVal = uprotocol::tools::Base64::base64encode((uint8_t *)origStr,
+                                                                origStrInvalidLen);
+    ASSERT_NE(encodedStrVal, encodedStrInVal);
+
+    size_t encodedStrInValLen = 5;
+    std::string encodedStr = uprotocol::tools::Base64::base64encode(origStr);
+    std::string decodedValStr = uprotocol::tools::Base64::base64decode((uint8_t *)encodedStrVal.data(),
+                                                                    encodedStrVal.length());
+    std::string decodedInValStr = uprotocol::tools::Base64::base64decode((uint8_t *)encodedStrVal.data(),
+                                                                       encodedStrInValLen);
+                                                                    
+    ASSERT_NE(decodedValStr, decodedInValStr);
+    ASSERT_EQ(decodedValStr.data(), std::string(origStr));
+};
+
+/**
+* Test Case: Tests base64 encode and decode when passing null data
+* Throw error in the log "Invalid data"
+*/
+TEST_F(Base64EncodeDecodeTests, null_data){
+   
+    static const uint8_t* origStr = nullptr;
+    size_t origStrLen = sizeof(origStr);
+
+    std::string encodedStrVal = uprotocol::tools::Base64::base64encode(origStr,
+                                                                       origStrLen);
+    
+    std::string decodedValStr = uprotocol::tools::Base64::base64decode(origStr,
+                                                                       encodedStrVal.length());
```

Comment:
Have assertions to check the expected 'decodedValStr'.

Reply:
ASSERT_EQ(decodedValStr.data(), std::string(origStr));

Reply:
Added